src/scales/scale.logarithmic.js

```diff
@@ -179,60 +179,76 @@ module.exports = function(Chart) {
 		},
 		getPixelForValue: function(value) {
 			var me = this;
+			var innerDimension = me.isHorizontal() ? me.width : me.height;
 			var start = me.start;
+			var end = me.end;
 			var newVal = +me.getRightValue(value);
 			var opts = me.options;
 			var tickOpts = opts.ticks;
-			var innerDimension, pixel, range;
+			var zero = me.isHorizontal() ? (tickOpts.reverse ? me.right : me.left) : (tickOpts.reverse ? me.top : me.bottom);
 
-			if (me.isHorizontal()) {
-				range = helpers.log10(me.end) - helpers.log10(start); // todo: if start === 0
-				if (newVal === 0) {
-					pixel = me.left;
-				} else {
-					innerDimension = me.width;
-					pixel = me.left + (innerDimension / range * (helpers.log10(newVal) - helpers.log10(start)));
-				}
-			} else {
-				// Bottom - top since pixels increase downward on a screen
-				innerDimension = me.height;
-				if (start === 0 && !tickOpts.reverse) {
-					range = helpers.log10(me.end) - helpers.log10(me.minNotZero);
-					if (newVal === start) {
-						pixel = me.bottom;
-					} else if (newVal === me.minNotZero) {
-						pixel = me.bottom - innerDimension * 0.02;
-					} else {
-						pixel = me.bottom - innerDimension * 0.02 - (innerDimension * 0.98 / range * (helpers.log10(newVal) - helpers.log10(me.minNotZero)));
-					}
-				} else if (me.end === 0 && tickOpts.reverse) {
-					range = helpers.log10(me.start) - helpers.log10(me.minNotZero);
-					if (newVal === me.end) {
-						pixel = me.top;
-					} else if (newVal === me.minNotZero) {
-						pixel = me.top + innerDimension * 0.02;
-					} else {
-						pixel = me.top + innerDimension * 0.02 + (innerDimension * 0.98 / range * (helpers.log10(newVal) - helpers.log10(me.minNotZero)));
-					}
-				} else if (newVal === 0) {
-					pixel = tickOpts.reverse ? me.top : me.bottom;
-				} else {
-					range = helpers.log10(me.end) - helpers.log10(start);
-					innerDimension = me.height;
-					pixel = me.bottom - (innerDimension / range * (helpers.log10(newVal) - helpers.log10(start)));
+			var pixel, range, valueOffset;
+
+			if (start === 0 || end === 0) { // 0 is in range
+				var min = Math.pow(10, Math.floor(helpers.log10(me.minNotZero)));
+				var max = tickOpts.reverse ? start : end;
+				range = helpers.log10(max) - Math.floor(helpers.log10(me.minNotZero));
+				var fontSize = helpers.getValueOrDefault(tickOpts.fontSize, Chart.defaults.global.defaultFontSize);
+				var diff = fontSize;
+				innerDimension -= diff;
```

Comment:
does subtracting `diff` work when the axis is horizontal and the labels are rotated less than 90 degrees?

Reply:
![screen-1](https://user-images.githubusercontent.com/19689271/29956223-6c8dbee4-8e9a-11e7-8e3b-ee1b60989560.jpg)

Horizontal chart looks like this.